clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -324,7 +324,8 @@ public void onSuccess(ClientResponse resp) {
                                             log.debug("Fetch {} at offset {} for partition {} returned fetch data {}",
                                                     isolationLevel, fetchOffset, partition, fetchData);
                                             completedFetches.add(new CompletedFetch(partition, fetchOffset, fetchData, metricAggregator,
-                                                    resp.requestHeader().apiVersion()));
+                                                    resp.requestHeader().apiVersion(), resp));
```

Comment:
[nit] Probably better to push resp.requestHeader().apiVersion() into the CompletedFetch constructor, instead of having an extra param here

Reply:
It is already part of the constructor. What do you mean?